python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -342,8 +343,8 @@ def toString(self, engine="Gromacs"):
     def correction(self):
         """Give the free energy of removing the restraint."""
         if self._restraint_type == "boresch":
-            K = (
-                _k_boltz * (_kcal_per_mol / _kelvin) / (_kj_per_mol / _kelvin)
+            K = _k_boltz.value() * (
```

Comment:
Is this intentional?

Reply:
```python
            K = _k_boltz.value() * (
                _kcal_per_mol / _kj_per_mol
            )  # Gas constant in kJ/mol/K
```

The `_kelvin` terms cancel. With Sire 2023.1.0 `_kboltz` now has units, so you need to call `.value()` to get it as a float.

Reply:
Thanks, but I was importing from sire legacy here so it is a bit worrying that these things got changed.

Reply:
I'm not sure if this is a change in the Python layer, not the C++ object itself. (Will check.) The code is versioned against Sire so it will only build against `sire~=2023.1.0`. I guess you are just copying the branch locally and testing in an existing environment.

Reply:
Thanks for the explanation. Sorry, I was under the impression that Sire changes but Sire.legacy retains the old behaviour forever. So this means that one should assume that Sire.legacy is updating as well?

Reply:
Yes, the objects in both APIs are identical, so it's a change in the C++ layer:

```python
from sire.units import k_boltz
type(k_boltz)
sire.legacy.Units._Units.GeneralUnit
```

```python
from sire.legacy.Units import k_boltz
type(k_boltz)
sire.legacy.Units._Units.GeneralUnit
```

From the changelog [here](https://sire.openbiosim.org/changelog.html#id1) all constants now are exposed with units, rather than being raw floats.

Reply:
I see. Thanks.

Reply:
Thanks. I will assume that the unit related things are changing in the sire.legacy.

Reply:
No problem. We always fix your code / tests when introducing changes, but I'll try to do a better job at clarifying what has changed and why during a PR so that there are no surprises.

Reply:
Sorry - `sire.legacy` tries to retain as much old behaviour as possible, but this won't be forever, and there are some small things that will change. As much as possible I am trying to minimise any changes, e.g. by having separate legacy and non-legacy objects, like `System`. The small things that change are to fix confusing or ambiguous behaviour. In this case, `k_boltz` was one of a small number of physical constants that didn't have units. This created issues when using the new `GeneralUnit` class, and so had to be updated.

As things have evolved I have realised that `legacy` is the wrong name, but I won't change it now ;-). All the C++ wrappers and C++ code is in legacy, and any new C++ code will also be added to legacy. The "new" API is fully python, and sits in the `sire` namespace. As we move forwards, the backwards-compatibility guarantees will be applied based on what is in the "new" API, and how it is described in the tutorials.

We are using the unit tests, plus larger integration tests in the [sire_bigtests](https://github.com/OpenBioSim/sire_bigtests) repo to minimise the risk of breakage caused by updates. Assuming good unit tests, we should (hopefully) be able to submit pull requests to fix any changes on your branch before you notice that there is a problem ;-)

Reply:
Thanks. I do understand that maintaining legacy and current version is quite difficult

Reply:
Sire hit 18 years old last weekend, so there is a lot of legacy ;-)

Reply:
Now is legally able to handle OH groups.